tests/backup/backup_kubevirt_test.go

```diff
@@ -3288,3 +3290,404 @@ var _ = Describe("{CustomBackupRestoreWithKubevirtAndNonKubevirtNS}", Label(Test
+			err = CreateRuleForVMBackup(unfreezeRuleName, allVMs, Unfreeze, ctx)
+			log.FailOnError(err, "Failed to create unfreeze rule %s for VMs - %v", unfreezeRuleName, allVMNames)
+			unfreezeRuleUid, err = Inst().Backup.GetRuleUid(BackupOrgID, ctx, unfreezeRuleName)
+			log.FailOnError(err, "Failed to get unfreeze rule uid for rule %s", unfreezeRuleName)
+		})
+
+		Step("Take backups during cordoning and draining the nodes in which VM is running", func() {
+			log.InfoD("Take backups during cordoning and draining the nodes in which VM is running")
+			k8sCore := core.Instance()
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			for _, appCtx := range scheduledAppContexts {
+				scheduledNamespace := appCtx.ScheduleOptions.Namespace
+				vms, err := GetAllVMsInNamespace(scheduledNamespace)
+				log.FailOnError(err, "Failed to get VMs deployed")
+				for _, vm := range vms {
+					createBackup := func(backupName string, namespace string) {
+						defer GinkgoRecover()
+						defer wg.Done()
+						appContextsToBackup := FilterAppContextsByNamespace(scheduledAppContexts, []string{namespace})
+						err = CreateVMBackupWithValidation(ctx, backupName, vms, SourceClusterName, backupLocationName, backupLocationUID, appContextsToBackup,
+							labelSelectors, BackupOrgID, sourceClusterUid, freezeRuleName, freezeRuleUid, unfreezeRuleName, unfreezeRuleUid, true)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Creation of VM backup and Validation of backup [%s]", backupName))
+						mu.Lock()
+						backupNames = append(backupNames, backupName)
+						backupNamespaceMap[backupName] = namespace
+						defer mu.Unlock()
+					}
+					log.Infof("Taking backup of VMs before cordoning the node")
+					wg.Add(1)
+					backupName := fmt.Sprintf("%s-%s-%v", "pre-cordon-backup", scheduledNamespace, time.Now().Unix())
+					go createBackup(backupName, scheduledNamespace)
+					wg.Wait()
+
+					log.Infof("Taking backup of VMs during cordoning the node")
+					wg.Add(1)
+					backupName = fmt.Sprintf("%s-%s-%v", "cordon-backup", scheduledNamespace, time.Now().Unix())
+					go createBackup(backupName, scheduledNamespace)
+					//Get the node where the vm is scheduled.
+					preNodeName, err := GetNodeOfVM(vm)
+					log.FailOnError(err, fmt.Sprintf("Failed to get nodename for the VM %s", vm.Name))
+					defer func() {
+						err = k8sCore.UnCordonNode(preNodeName, 1*time.Minute, 5*time.Second)
+						dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying uncordoning the node %s", preNodeName))
+					}()
+					err = k8sCore.CordonNode(preNodeName, 1*time.Minute, 5*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Verifying cordoning the node %s", preNodeName))
+					vmPod, err := GetVirtLauncherPodObject(vm)
+					log.FailOnError(err, fmt.Sprintf("Failed to get pod for the VM %s", vm.Name))
+					err = k8sCore.DrainPodsFromNode(preNodeName, []corev1.Pod{*vmPod}, 30*time.Minute, 30*time.Second)
```

Comment:
we are not waiting for backup to be completed, hence operation will executed in parallel.
sample log snippet for both operation in parallel.

`12:48:48  2024-04-16 07:18:46 +0000:[INFO] [{KubevirtVMMigrationTest}] virt-launcher pod found for vm [vm-cirros-containerdisk] is [virt-launcher-vm-cirros-containerdisk-sp7ms]

12:48:48  2024/04/16 07:18:46 DoRetryWithTimeout - Error: {Operation cannot be fulfilled on nodes "ip-10-13-156-118.pwx.purestorage.com": the object has been modified; please apply your changes to the latest version and try again}, Next try in [30s], timeout [30m0s]

12:48:48  2024/04/16 07:18:46 DoRetryWithTimeout - Error: {backup status for [cordon-backup-kubevirt-cirros-live-migration-93442-0-93442-04-16-07h16m56s-1713251925] expected was [[Success]] but got [Pending] because of []}, Next try in [30s], timeout [40m0s]

12:49:21  2024/04/16 07:19:16 DoRetryWithTimeout - Error: {pod kubevirt-cirros-live-migration-93442-0-93442-04-16-07h16m56s:virt-launcher-vm-cirros-containerdisk-sp7ms (c695dbb2-be0c-4617-9f63-61261efc639e) still present in the system}, Next try in [5s], timeout [30m0s]

12:49:21  2024/04/16 07:19:16 DoRetryWithTimeout - Error: {backup status for [cordon-backup-kubevirt-cirros-live-migration-93442-0-93442-04-16-07h16m56s-1713251925] expected was [[Success]] but got [InProgress] because of [Application resources backup is in progress]}, Next try in [30s], timeout [40m0s]

12:49:21  2024/04/16 07:19:21 DoRetryWithTimeout - Error: {pod kubevirt-cirros-live-migration-93442-0-93442-04-16-07h16m56s:virt-launcher-vm-cirros-containerdisk-sp7ms (c695dbb2-be0c-4617-9f63-61261efc639e) still present in the system}, Next try in [5s], timeout [30m0s]

12:49:27  2024/04/16 07:19:26 DoRetryWithTimeout - Error: {pod kubevirt-cirros-live-migration-93442-0-93442-04-16-07h16m56s:virt-launcher-vm-cirros-containerdisk-sp7ms (c695dbb2-be0c-4617-9f63-61261efc639e) still present in the system}, Next try in [5s], timeout [30m0s]`